be/src/exec/hdfs_scanner_orc.cpp

```diff
@@ -372,7 +385,46 @@ Status HdfsOrcScanner::do_open(RuntimeState* runtime_state) {
         _src_slot_descriptors.emplace_back(column.slot_desc);
         src_slot_index++;
     }
+    return Status::OK();
+}
 
+Status HdfsOrcScanner::do_open(RuntimeState* runtime_state) {
+    // create wrapped input stream.
+    RETURN_IF_ERROR(open_random_access_file());
+    if (_input_stream == nullptr) {
+        _input_stream = std::make_unique<ORCHdfsFileStream>(_file.get(), _file->get_size().value(),
+                                                            _shared_buffered_input_stream.get());
+        _input_stream->set_lazy_column_coalesce_counter(_scanner_ctx.lazy_column_coalesce_counter);
+        _input_stream->set_app_stats(&_app_stats);
+    }
+    ORCHdfsFileStream* orc_hdfs_file_stream = _input_stream.get();
+
+    // create orc reader on this input stream.
+    SCOPED_RAW_TIMER(&_app_stats.reader_init_ns);
+    std::unique_ptr<orc::Reader> reader;
+    try {
+        orc::ReaderOptions options;
+        options.setMemoryPool(*getOrcMemoryPool());
+        reader = orc::createReader(std::move(_input_stream), options);
+    } catch (std::exception& e) {
+        auto s = strings::Substitute("HdfsOrcScanner::do_open failed. reason = $0", e.what());
+        LOG(WARNING) << s;
+        if (errno == ENOENT || s.find("404") != std::string::npos) {
```

Comment:
Will error be changed after calling LOG?

Reply:
No. Why error will be changed after calling LOG?

Reply:
sorry i mean errno

Reply:
I'm not quite sure if `errno == ENOENT` really works in this branch.

actually what im sure is that we DO peek the error message to find "404" in that.